bundles/org.dataflowanalysis.analysis.converter/src/org/dataflowanalysis/analysis/converter/PalladioConverter.java

```diff
@@ -0,0 +1,184 @@
+        }
+    }
+
+    // A pin is equivalent if the same parameters are passed
+    private Pin findOrCreateOutputPin(Node source, String parameters) {
+        Optional<Pin> optPin = source.getBehaviour().getOutPin().stream().filter(p -> p.getEntityName().equals(parameters)).findAny();
+        if (optPin.isPresent()) {
+            return optPin.get();
+        }
+        Pin pin = datadictionaryFactory.eINSTANCE.createPin();
+        pin.setEntityName(parameters);
+        source.getBehaviour().getOutPin().add(pin);
+        return pin;
+    }
+
+    // A pin is equivalent if the same parameters are passed
+    private Pin findOrCreateInputPin(Node dest, String parameters) {
+        Optional<Pin> optPin = dest.getBehaviour().getInPin().stream().filter(p -> p.getEntityName().equals(parameters)).findAny();
+        if (optPin.isPresent()) {
+            return optPin.get();
+        }
+        Pin pin = datadictionaryFactory.eINSTANCE.createPin();
+        pin.setEntityName(parameters);
+        dest.getBehaviour().getInPin().add(pin);
+        return pin;
+    }
+
+    private Node getOrCreateDFDNode(AbstractPCMActionSequenceElement<? extends Entity> pcmASE) {
+        Node dfdNode = dfdNodeMap.get(pcmASE.getElement());
+
+        if (dfdNode == null) {
+            dfdNode = createCorrespondingDFDNode(pcmASE);
+            dfdNodeMap.put(pcmASE.getElement(), dfdNode);
+        }
+
+        addNodeCharacteristicsToNode(dfdNode, pcmASE.getAllNodeCharacteristics());
+
+        return dfdNode;
+    }
+
+    private Node createCorrespondingDFDNode(AbstractPCMActionSequenceElement<? extends Entity> pcmASE) {
+        Node node;
+
+        if (pcmASE instanceof UserActionSequenceElement<?>) {
+            node = dataflowdiagramFactory.eINSTANCE.createExternal();
+        } else if (pcmASE instanceof SEFFActionSequenceElement<?>) {
+            node = dataflowdiagramFactory.eINSTANCE.createProcess();
+        } else {
+            return null;
+        }
+
+        Behaviour behaviour = datadictionaryFactory.eINSTANCE.createBehaviour();
+        node.setEntityName(pcmASE.getElement().getEntityName());
+        node.setId(pcmASE.getElement().getId());
+        node.setBehaviour(behaviour);
+        dataDictionary.getBehaviour().add(behaviour);
+        dataFlowDiagram.getNodes().add(node);
+        return node;
+    }
+
+    private void addNodeCharacteristicsToNode(Node node, List<CharacteristicValue> charValues) {
+        for (CharacteristicValue charValue : charValues) {
+            Label label = getOrCreateDFDLabel(charValue);
+            if (!node.getProperties().contains(label)) {
+                node.getProperties().add(label);
+            }
+        }
+    }
+
+    private Label getOrCreateDFDLabel(CharacteristicValue charValue) {
+        LabelType type = dataDictionary.getLabelTypes().stream().filter(f -> f.getEntityName().equals(charValue.getTypeName())).findFirst()
+                .orElse(null);
```

Comment:
Please do not use or else to extract the null value from an optional. Provide the value from the if case as the value for the method instead